### DISCUSSION TITLE
selecting multiple fields json to csv

### DISCUSSION THREAD

**Discussion by john-mx** [TomWright/dasel#335] [Q&A]
I want to extract a few fields from a json file and save the results to csv (or excel).  There's an old article on Medium that seems to get multiple files but it's beyond me.  I don't know if it's related but I also can't figure out how to string multiple commands.  Docs say "use a ." but there's no example.  
Want to `dasel -f my.json -s 'all().col1' 'all().col3' -w csv > my.csv`
Is there any way to do that?  
And how to you use the dot linkage? I don't understand it.

**Comment by TomWright** [comment]
You can do that like this:
```
echo '[{"col1":"x","col2":"y"},{"col1":"1","col2":"2"}]' | dasel -r json -w csv 'all().mapOf(col 1 header,col1,col 2 header,col2).merge()'
col 1 header,col 2 header
x,y
1,2
```

Steps:
- Grab all items from the JSON array using `all()`
- Use those items to create a new object/map using `mapOf()`
- Merge all items back into an array using `merge()`.
- The `.` is the separator between each func.

**Reply by rubenvereecken** [reply]
@TomWright if I wanted to keep the column names the same, is there a way to not specify them?

Right now I've got something like `mapOf(id,id,name,name)` and it's highly repetitive (and, I'm worried, error-prone).

**Reply by TomWright** [reply]
> @TomWright if I wanted to keep the column names the same, is there a way to not specify them?
> 
> Right now I've got something like `mapOf(id,id,name,name)` and it's highly repetitive (and, I'm worried, error-prone).
> 
> 

Right now, no. I'd be happy to support something to do this though
